Sprint2 ord/ods naming systems + not-present codesystems

APageNHS opened the pull request:
Currently, the naming systems are failing validation due to url, @ryma2fhir can we do anything about that

CodeSystems may end up populated, ORD/ODS team to confirm by start of sprint 2 call

APageNHS committed:
NamingSystems and basic not-present CodeSystems

APageNHS committed:
Switch OrgRole to new codesystem url

APageNHS committed:
Fix duff id/name

APageNHS committed:
Remove spaces from name

APageNHS committed:
Remove spaces from name

APageNHS committed:
Set naming systems to draft

APageNHS committed:
Merge 03cecfc765e6464185a9de120733d4537a30c87a into a64458dfd7d72f947a170d1c71cca1eb1a8ad3f3

ryma2fhir committed:
GitHub Action: Find and Replace

johngeo commented:
Looks ok, but I can see a Validation failure: "IOPS-FHIR-QualityControlChecker-call / call-workflow / build (push) Failing after 14s".  It looks like this is because the validator isn't recognising the URL: https://ord.nhs.uk/...  Can the validator be updated to accept this URL?

KazeemHamzat reviewed (commented):
Any reason why version number of CodeSystems and ValueSet is 0.0.1? Different from other assets in NHSE IG starting with 1.0.0

APageNHS commented:
> Any reason why version number of CodeSystems and ValueSet is 0.0.1? Different from other assets in NHSE IG starting with 1.0.0

This was discussed on one of the calls last week, were it was suggested that we should start draft assets at 0.0.1, and up the version to 1.0.0 when they are made active

ryma2fhir commented:
> @APageNHS Looks ok, but I can see a Validation failure: "IOPS-FHIR-QualityControlChecker-call / call-workflow / build (push) Failing after 14s". It looks like this is because the validator isn't recognising the URL: https://ord.nhs.uk/... Can the validator be updated to accept this URL?



> @APageNHS Looks ok, but I can see a Validation failure: "IOPS-FHIR-QualityControlChecker-call / call-workflow / build (push) Failing after 14s". It looks like this is because the validator isn't recognising the URL: https://ord.nhs.uk/... Can the validator be updated to accept this URL?

IF @APageNHS reviews https://github.com/NHSDigital/IOPS-FHIR-Test-Scripts/pull/32 then the validator will check this uri

ryma2fhir commented:
I see the other CodeSystems in the NHSE IG start with `https://fhir.nhs.uk/England/CodeSystem/` whereas these start with `https://ord.nhs.uk/CodeSystem/` without the `England` in between. Is this expected?

APageNHS commented:
> I see the other CodeSystems in the NHSE IG start with `https://fhir.nhs.uk/England/CodeSystem/` whereas these start with `https://ord.nhs.uk/CodeSystem/` without the `England` in between. Is this expected?

These are the URLs that have been requested by ORD/ODS, so yes. And I should have noticed that they were different to the rule you added to the qualitychecker, thats my fault

ryma2fhir commented:
Next issue. The url suffix doesn't follow the naming convention, e.g.
`CodeSystem-England-ORDRecordUseType` url is `https://ord.nhs.uk/CodeSystem/record-use-type`. I'd expect it to be `https://ord.nhs.uk/CodeSystem/ORDRecordUseType`

APageNHS commented on NamingSystems/ods-organization-code-namingsystem.xml:
```diff
@@ -0,0 +1,36 @@
+<NamingSystem xmlns="http://hl7.org/fhir">
+  <id value="ods-organization-code-namingsystem" />
```
This replicates one in UK Core, so probably not needed, need to check with KS

APageNHS commented on NamingSystems/ord-record-class-namingsystem.xml:
```diff
@@ -0,0 +1,24 @@
+<NamingSystem xmlns="http://hl7.org/fhir">
```
Expectation is that ORD/ODS will request NOT to have namingsystems next week

APageNHS commented on codesystems/CodeSystem-England-ORDRelationship.xml:
```diff
@@ -0,0 +1,22 @@
+<CodeSystem xmlns="http://hl7.org/fhir">
+  <id value="England-ORDRelationship" />
+  <url value="https://ord.nhs.uk/CodeSystem/relationship" />
```
Expectation is ORD/ODS will backtrack on URLs next week

APageNHS replied:
Backtrack was on naming systems, url to stay, but may want to revert from ord to ods during review period

digitalhealthmartin commented on NamingSystems/ods-organization-code-namingsystem.xml:
All NamingSystem files (this and the others) should follow the format of NamingSystem-England-[Name].xml

digitalhealthmartin commented on NamingSystems/ods-organization-code-namingsystem.xml:
```diff
@@ -0,0 +1,36 @@
+<NamingSystem xmlns="http://hl7.org/fhir">
+  <id value="ods-organization-code-namingsystem" />
+  <name value="ODSOrganisationCode" />
```
Name should be in the format of England[Name]

APageNHS replied:
Namingsystem removed

digitalhealthmartin commented on NamingSystems/ods-organization-code-namingsystem.xml:
```diff
@@ -0,0 +1,36 @@
+  <date value="2023-08-01" />
+  <publisher value="NHS England" />
+  <contact>
+        <name value="Organisation Data Service" />
```
This needs a discussion. Ideally the IOPS team contact details should be here, or we have both contact details here.

APageNHS replied:
Agree, probably worth documenting that in UK Core and NHSe design approach for going forward. NamingSystems, removed, but have used this principle, and added the ODS contact details to their codesystems.

digitalhealthmartin commented on NamingSystems/ods-organization-code-namingsystem.xml:
```diff
@@ -0,0 +1,36 @@
+        </telecom>
+    </contact>
+  <responsible value="Organisation Data Service" />
+  <description value="The identifier system for an Organisation registered with the Organisation Data Service (ODS)." />
```
I think Organisation should either be 'organisation' (small o and s) or 'Organization' (big o and little z for FHIR asset). Same as line 19 too.

APageNHS replied:
Good cactch, this will need doing in UK Core

digitalhealthmartin reviewed (commented):
Made some prelim comments but then started to notice comments that suggest this is still WIP and not ready for review. Will continue review once confirmed this is actually ready for review.

APageNHS committed:
Merge branch 'stu1_develop' into sprint2/ODS_NamingSystems

APageNHS committed:
Remove namingsystems, tweak codesystems to have ODS contact details, and url for the codes

APageNHS committed:
Merge 8de6a85d9ad9b190059bf0279d9e49dda2917076 into d321a1338318bf97467d338a6659526ae497918f

ryma2fhir committed:
GitHub Action: Find and Replace

APageNHS commented on codesystems/CodeSystem-England-ORDRecordUseType.xml:
```diff
@@ -0,0 +1,31 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="A set of codes that define the intended use of a record. Full records are complete, whereas skeleton/partial records are provided for reference use only. Record use type is distinct from Status and Record Class.&#13;&#13;For more information and an up to date list see: [Organisation Data Service CodeSet](https://digital.nhs.uk/services/organisation-data-service/codesystems). ...
```
NOTE: URL will not resolve ATM

digitalhealthmartin replied:
Mentioned broken URL in previous assets.

digitalhealthmartin replied:
Feel like either first/second sentence should specifically refer to ODS/ORD use-case as it's only until the final sentence where it states an ODS URL.

APageNHS replied:
Added reference to ODS to first sentence

APageNHS commented on codesystems/CodeSystem-England-ORDRecordUseType.xml:
```diff
@@ -0,0 +1,31 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <contact>
```
Added ODS contact details, which would have otherwise been in the NamingSystem

APageNHS commented:
Updated:

NamingSystems removed
ODS contact details added to CodeSystems
Swapped version back to 1.0.0
Added the destination url that ORD/ODS have provided, where their codes will be published

Note - Potential changes that will come during review period:
1) Period Type codes to be upper case for the first letter, as this is the code already in use
2) Claification of ORD / ODS usage, probably going back to ODS

digitalhealthmartin commented on codesystems/CodeSystem-England-ORDOrganisationRole.xml:
```diff
@@ -0,0 +1,31 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="A set of codes that define the organisation type (primary role, mandatory) and any non primary roles that give added information for an organisation/site's function, for the Organisation Data Service (ODS).&#13;&#13;For more information and an up to date list see: [Organisation Data Service CodeSet](https://digital.nhs.uk/services/organisation-data-service/codesystems)." />
```
Description contains info on primary/non primary but believe this is no longer required?

digitalhealthmartin replied:
Also URL gets 404

APageNHS replied:
Removed reference to primary. As discussed, site is not live and out of our control, but this is the url agreed to be used within ORD/ODS.

digitalhealthmartin commented on codesystems/CodeSystem-England-ORDRecordClass.xml:
```diff
@@ -16,7 +16,16 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="A set of codes that define the high level type / class of record published by Organisation Data Service (ODS).&#13;&#13;For more information and an up to date list see: [Organisation Data Service CodeSet](https://digital.nhs.uk/services/organisation-data-service/codesystems)." />
```
URL has 404

APageNHS replied:
As discussed, site is not live and out of our control, but this is the url agreed to be used within ORD/ODS.

digitalhealthmartin commented on codesystems/CodeSystem-England-ORDRecordUseType.xml:
```diff
@@ -0,0 +1,31 @@
+  <url value="https://ord.nhs.uk/CodeSystem/record-use-type" />
+  <version value="1.0.0" />
+  <name value="EnglandORDRecordUseType" />
+  <title value="England ORD RecordUseType" />
```
Title should be 'England ORD Record Use Type'

APageNHS replied:
Changed as requested

digitalhealthmartin commented on valuesets/ValueSet-England-OrganisationRole.xml:
```diff
@@ -1,11 +1,11 @@
   <id value="England-OrganisationRole" />
   <url value="https://fhir.nhs.uk/England/ValueSet/England-OrganisationRole" />
-  <version value="1.0.0" />
+  <version value="0.0.1" />
```
Please change to 1.0.0

APageNHS replied:
Changed as requested

digitalhealthmartin commented on valuesets/ValueSet-England-TypedDateTime.xml:
Within contains, referencing URL https://fhir.nhs.uk/England/CodeSystem/England-ORDDateTime however other CodeSystems relating to ORD use case have used their own custom URL formatting. Is this correct or does this (and the CodeSystem) URL need updating?

APageNHS replied:
ORD are happy for us to take responsibility for this url.

APageNHS committed:
Tweaks following review

APageNHS committed:
Merge fixes

APageNHS committed:
Merge f673f4aeeed175fd6d8f7a3eb81a35bce3f2e351 into d321a1338318bf97467d338a6659526ae497918f

ryma2fhir committed:
GitHub Action: Find and Replace

digitalhealthmartin commented on codesystems/CodeSystem-England-ORDOrganisationRole.xml:
```diff
@@ -0,0 +1,31 @@
+      <rank value="1" />
+    </telecom>
+  </contact>
+  <description value="A set of codes that define the organisation roles that give added information for an organisation/site's function, for the Organisation Data Service (ODS).&#13;&#13;For more information and an up to date list see: [Organisation Data Service CodeSet](https://digital.nhs.uk/services/organisation-data-service/codesystems)." />
```
There is comma towards the end of the sentence which probably should be removed. Also the bit after the comma doesn't quite make sense so may want to reword it a bit.

APageNHS replied:
Reworded based on input from ODS, but at under 24 hours before the sprint, a comma in a draft profile should not be hold this up.

digitalhealthmartin reviewed (changes requested):
Just minor rework on description

APageNHS committed:
reword

APageNHS committed:
fix

APageNHS committed:
Merge e5ca362fd8e547a36fea0bc31c0ccd8fced9aafa into d321a1338318bf97467d338a6659526ae497918f

ryma2fhir committed:
GitHub Action: Find and Replace
